src/qibolab/result.py

```diff
@@ -5,97 +5,186 @@
 import numpy as np
 import numpy.typing as npt
 
-ExecRes = np.dtype([("i", np.float64), ("q", np.float64)])
+iq_typing = np.dtype([("i", np.float64), ("q", np.float64)])
```

Comment:
```suggestion
iq = np.dtype([("i", np.float64), ("q", np.float64)])
```

I would even avoid the `_typing` suffix, it is redundant.

If `iq` seems too ambiguous (which I don't believe, in this context should be perfectly fine), and in lack of any other better name, at least shorten to `iqtype`.
It is a `dtype` (not a `dtyping`, `typing` is an action and the name of some modules that are performing the action of giving types), and avoiding underscores is considered Pythonic, when readability is not affected.

> with words separated by underscores as necessary to improve readability
> https://peps.python.org/pep-0008/#function-and-variable-names

(in other languages is more idiomatic to always separate, but not Python)